### PR TITLE
Add HighestAttestations function

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Feature 

**What does this PR do? Why is it needed?**
This PR is part of #8331, to maintain functionality in the new slasher. This new DB function takes in an array of ValidatorIndexs and returns the highest attestation data from them.

### CODE REVIEW

File: beacon-chain/db/kv/slasher.go [hunk 1]
```diff
@@ -353,11 +355,62 @@ func (s *Store) PruneAttestations(ctx context.Context, currentEpoch types.Epoch,
...
+	}
+
+	// Sort indices to keep DB interactions short.
+	sort.SliceStable(indices, func(i, j int) bool {
+		return uint64(indices[i]) < uint64(indices[j])
+	})
+
+	history := make([]*slashpb.HighestAttestation, 0, len(indices))
+
+	err := s.db.Update(func(tx *bolt.Tx) error {
```
Comment (rauljordan) [inline beacon-chain/db/kv/slasher.go:374]:
```suggestion
	err := s.db.View(func(tx *bolt.Tx) error {
```

File: beacon-chain/db/kv/slasher_test.go [hunk 2]
```diff
@@ -482,6 +483,139 @@ func TestStore_PruneAttestations(t *testing.T) {
...
+
+	allIndices := make([]types.ValidatorIndex, valsPerAtt*count)
+	for i := 0; i < count; i++ {
+		indicesForAtt := make([]types.ValidatorIndex, valsPerAtt)
+		for r := 0; r < valsPerAtt; r++ {
+			indicesForAtt[r] = types.ValidatorIndex(atts[i].IndexedAttestation.AttestingIndices[r])
+		}
+		allIndices = append(allIndices, indicesForAtt...)
+	}
+	b.ReportAllocs()
```
Comment (rauljordan) [inline beacon-chain/db/kv/slasher_test.go:612]: We need to only bench starting from here. Do b.StopTimer() at the top of the func Benchmark and b.StartTimer right before this for loop

**Review by rauljordan — DISMISSED** [review]

File: beacon-chain/db/kv/slasher.go [hunk 3]
```diff
@@ -353,11 +355,62 @@ func (s *Store) PruneAttestations(ctx context.Context, currentEpoch types.Epoch,
...
+	sort.SliceStable(indices, func(i, j int) bool {
+		return uint64(indices[i]) < uint64(indices[j])
+	})
+
+	history := make([]*slashpb.HighestAttestation, 0, len(indices))
+
+	err := s.db.View(func(tx *bolt.Tx) error {
+		attBkt := tx.Bucket(attestationRecordsBucket)
+		for i := 0; i < len(indices); i++ {
+			encIdx, err := indices[i].MarshalSSZ()
```
Comment (prestonvanloon) [inline beacon-chain/db/kv/slasher.go:377]: Can you do these outside of the view transaction? Anything to reduce the open View transaction would be ideal

Comment (prestonvanloon) [reply beacon-chain/db/kv/slasher.go:377]: Specifically, set it up like this:
```go
encodedIndices := make([][]byte, len(indices))
for i, idx := range indices {
  encodedIndices[i], err = idx.MarshalSSZ()
  if err != nil {
   return err
   }
}
```

Then just use encodedIndices

Comment (rauljordan) [reply beacon-chain/db/kv/slasher.go:377]: good point

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**